build-system/tasks/e2e/amp-driver.js

```diff
@@ -43,6 +44,21 @@ const EnvironmentBehaviorMap = {
       return `http://localhost:8000/examples/viewer.html#href=${url}`;
     },
   },
+
+  [AmpdocEnvironment.SHADOW_DEMO]: {
+    async ready(controller) {
+      // TODO(cvializ): this is a HACK
+      // There should be a better way to detect that the shadowdoc is ready.
+      const shadowHost = await controller.findElement(
+          '.amp-doc-host[style="visibility: visible;"]');
```

Comment:
Would do `'.amp-doc-host[style*="visibility: visible;"]'` to be robust against other styles being added in the future.

Reply:
no : ( I'll add a more robust mechanism in a followup PR